modules/bash-commons/src/aws.sh

```diff
@@ -60,6 +60,144 @@ function aws_get_instance_tags {
     --filters "Name=resource-type,Values=instance" "Name=resource-id,Values=$instance_id"
 }
 
+# Return the value of the $tag_key for the given EC2 Instance $instance_id
+function aws_get_instance_tag_val {
+  local -r tag_key="$1"
+  local -r instance_id="$2"
+  local -r instance_region="$3"
+
+  local tags
+  tags=$(aws_get_instance_tags "$instance_id" "$instance_region")
+
+  local tag_val
+  tag_val=$(echo "$tags" | jq -r ".Tags[] | select( .ResourceType == \"instance\" and .Key == \"$tag_key\") | .Value")
+
+  echo "$tag_val"
+}
+
+# Return all ENIs attached to the current EC2 Instance
+function aws_get_enis_for_instance {
+  local -r instance_id="$1"
+  local -r aws_region="$2"
+
+  aws ec2 describe-network-interfaces --region "$aws_region" --filters "Name=attachment.instance-id,Values=$instance_id"
+}
+
+# Find all Elastic Network Interfaces (ENIs) that have a matching $tag_key=$tag_value
+function aws_get_enis_for_tag {
+  local -r tag_key="$1"
+  local -r tag_value="$2"
+  local -r aws_region="$3"
+
+  aws ec2 describe-network-interfaces --region "$aws_region" --filters "Name=tag:$tag_key,Values=$tag_value"
+}
+
+# Given the $network_interfaces_output, return the ID of the ENI at the given $eni_index (zero-based)
+#
+# Example:
+#   network_interfaces_output=$(aws_get_enis_for_this_ec2_instance)
+#   aws_get_eni_id "$network_interfaces_output" 0
+function aws_get_eni_id {
+  local -r network_interfaces_output="$1"
+  local -r eni_device_index="$2"
+
+  assert_not_empty "network_interfaces_output" "$network_interfaces_output" "Value returned from AWS API describe-network-interfaces output"
+  assert_not_empty "eni_device_index" "$eni_device_index"
```

Comment:
Hm, this file does not `source` the `assert.sh` file. To use these, you'd probably have to, or document in the method that the user must do this.

Reply:
I'll source both. I don't think there are many files that include aws.sh without log.sh and assert.sh already, so it's probably safe to source them for all.